### PR TITLE
feature(unlock-app): User events' listing

### LINKED ISSUE #11666: [Feature] Allow user to see previously created events
Currently, user can only see the "events" type of lock via the dashboard. I'd recommend to allow user view their previously created locks in the events use case page, the event list should use same components from the dashboard. 

prerequisite: user has to connect the wallet

figma: https://www.figma.com/file/glmh5GX316O4TGiwaPyC8f?node-id=331:6386#420201413

image

### PR DESCRIPTION
**Pull request by 0xTxbi**
# Description
This PR introduces a new page that displays all events managed by a user.

# Issues
Fixes #11666 
Refs #11666

## ScreenGrab
Screenshot 2024-10-03 at 14 06 15

## Flow
[Loom](https://www.loom.com/share/bf0687d8411b4556afbf6713516f2000?sid=4a18a2a4-8465-495d-b126-71a9b231321d)

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/List/elements/EventCard.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+const EventImage = ({ imageUrl }: EventImageProps) => {
+  return (
+    <div className="relative block overflow-hidden rounded-2xl h-20 w-20 group">
+      <img alt="logo" className="object-cover h-full aspect-1" src={imageUrl} />
+    </div>
+  )
+}
+
+export const EventCard = ({ event }: EventCardProps) => {
+  const { name, lockAddress, network, slug, ticket, image: imageUrl } = event
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/List/elements/EventCard.tsx:41]: The event should have a checkout config and we should use that one?

File: unlock-app/src/components/interface/locks/List/elements/EventCard.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            >
+              <div className="flex items-center gap-2">
+                <CryptoIcon symbol={lockData?.currencySymbol} />
+                <span className="overflow-auto text-ellipsis">
+                  {typeof lockData?.keyPrice !== 'string' ||
+                  parseFloat(lockData?.keyPrice) <= 0 ? (
+                    '0'
+                  ) : (
+                    <PriceFormatter price={lockData?.keyPrice} precision={2} />
+                  )}
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/List/elements/EventCard.tsx:101]: I am not sure the price of ticket is useful info. As an organizer I probably care more about the number of folks who have RSVP'ed. wdt?

**Review by julien51 — CHANGES_REQUESTED** [review]
Please cehck that you are lot loading the locks from the event...

File: unlock-app/src/components/interface/locks/List/elements/EventCard.tsx (hunk 1 above)
Comment (0xTxbi) [reply unlock-app/src/components/interface/locks/List/elements/EventCard.tsx:41]: alright!

File: unlock-app/src/components/interface/locks/List/elements/EventCard.tsx (hunk 2 above)
Comment (0xTxbi) [reply unlock-app/src/components/interface/locks/List/elements/EventCard.tsx:101]: same thought here!

**Comment by julien51** [comment]
I found the Github issue! https://github.com/unlock-protocol/unlock/issues/11666

**Comment by 0xTxbi** [comment]
> I found the Github issue! #11666

okay awesome!

File: unlock-app/src/hooks/useRsvp.ts [hunk 3]
```diff
@@ -33,3 +49,37 @@ export const useRsvp = ({ lockAddress, network }: Options) => {
     retry: 2,
   })
 }
+
+// fetch the number of RSVPs for an event
+export const useEventRSVP = ({
+  lockAddress,
+  network,
+  eventEndDate,
+}: UseEventRSVPProps) => {
```
Comment (julien51) [inline unlock-app/src/hooks/useRsvp.ts:58]: We should probably check for _all_ locks for an event for completeness. So here this function would use the checkout Config and loop over each lock.

**Review by julien51 — CHANGES_REQUESTED** [review]
One more change. An event might have several locks ( see https://app.unlock-protocol.com/event/farcon for example) so we need to count all keys across all locks.

File: unlock-app/src/hooks/useRsvp.ts (hunk 3 above)
Comment (0xTxbi) [reply unlock-app/src/hooks/useRsvp.ts:58]: true, good catch! will fix!

**Comment by 0xTxbi** [comment]
> One more change. An event might have several locks ( see https://app.unlock-protocol.com/event/farcon for example) so we need to count all keys across all locks.

got it!

**Review by julien51 — APPROVED** [review]
LGTM!

**Pull request merged**